packages/core/src/State.ts

```diff
@@ -21,6 +21,7 @@ import { StateNode } from './StateNode';
 import { getMeta, nextEvents } from './stateUtils';
 import { initEvent } from './actions';
 import { IS_PRODUCTION } from './environment';
+import { DefaultTypegenMeta, TypegenIsActive, TypegenMeta } from '.';
```

Comment:
It's best to avoid cycles in the module graph
```suggestion
import { DefaultTypegenMeta, TypegenIsActive, TypegenMeta } from './types';
```